src/verbs/queue_pair.rs

```diff
@@ -176,13 +177,22 @@ pub trait QueuePair {
     type Guard<'g>: PostSendGuard
     where
         Self: 'g;
-    fn start_post_send<'qp, 'g>(&'qp mut self) -> Self::Guard<'g>
-    where
-        'qp: 'g;
+    fn start_post_send(&mut self) -> Self::Guard<'_>;
+
+    fn start_post_recv(&mut self) -> PostRecvGuard<'_> {
```

Comment:
`post_recv` or `post_receive`? There are `recv`s and `receive`s in our project, do you think we should align the name?

Reply:
I think `recv` is a valid name to represent an I/O operation. There are many examples in tokio that use this term as well. https://docs.rs/tokio/latest/tokio/?search=recv

> For original `ibv_post_send` and `ibv_post_recv` APIs, would it be a correct usage for users to `post_send` and `post_recv` on different threads at the same time for a single QP? Should it be possible for users to hold `PostSendGuard` and `PostRecvGuard` at the same time?

Yes, I think that is a valid usage, although it is not commonly used. We may implement an extension in the future which splits `QueuePair` to a pair of `SendQueue` and `RecvQueue` so that users can create guard on them simultaneously.